release/release-notes/data-prepper.release-notes-2.10.0.md

```diff
@@ -0,0 +1,40 @@
+## 2024-10-14 Version 2.10.0
```

Comment:
```suggestion
## 2024-10-15 Version 2.10.0
```

Please update the date for today's release

Reply:
Thanks @dlvenable. I have updated the date.